File: scripts/youngcha/distribution/simpletransmission.js
const rotL = require("libraries/rotpowerlib");

const simpleTrans = rotL.torqueExtend(Block, Building, "simple-transmission",rotL.baseTypes.torqueTransmission, {
	
	load(){
		this.super$load();
		this.topsprite = [];
		this.topsprite = [Core.atlas.find(this.name + "-top1"),Core.atlas.find(this.name + "-top2")];
		this.overlaysprite = [Core.atlas.find(this.name + "-overlay1"),Core.atlas.find(this.name + "-overlay2")];
		this.moving = [Core.atlas.find(this.name + "-moving1"),Core.atlas.find(this.name + "-moving2"),Core.atlas.find(this.name + "-moving3")];
		this.movingtest = Core.atlas.find(this.name + "-moving");
		this.base = Core.atlas.find(this.name + "-bottom");
		this.mbase = Core.atlas.find(this.name + "-mbase");
		
	},
	
},{
	
	updatePre()
	{
		this.getGraphConnector("torque graph").setInertia(25);
		this.getGraphConnector("torque graph").setFriction(0.05);
	},
	
	draw() {
		let torquegraph = this.getGraphConnector("torque graph");
		let fixedrot = ((this.rotdeg()+90)%180)-90;
		let variant = ((this.rotation+1)%4>=2)?1:0;
		Draw.rect(simpleTrans.base, this.x, this.y, 0);
		Draw.rect(simpleTrans.mbase, this.x, this.y, this.rotdeg());
		//speeeeeeen
		let offset = rotL.dirs[(this.rotation+1)%4];
		let ox = offset.x*4;
		let oy = offset.y*4;
		rotL.drawRotRect(simpleTrans.moving[0], this.x+ox, this.y+oy, 16, 18/4, 18/4, fixedrot, torquegraph.getRotationOf(0), torquegraph.getRotationOf(0)+180);
		rotL.drawRotRect(simpleTrans.moving[0], this.x+ox, this.y+oy, 16, 18/4, 18/4, fixedrot, torquegraph.getRotationOf(0)+180, torquegraph.getRotationOf(0)+360);
		
		rotL.drawRotRect(simpleTrans.moving[1], this.x+ox*-0.125, this.y+oy*-0.125, 16, 18/4, 18/4, fixedrot, 360-(torquegraph.getRotationOf(0)), 360-(torquegraph.getRotationOf(0)+180));
		rotL.drawRotRect(simpleTrans.moving[1], this.x+ox*-0.125, this.y+oy*-0.125, 16, 18/4, 18/4, fixedrot, 720-(torquegraph.getRotationOf(0)+180), 720-(torquegraph.getRotationOf(0)+360));
		
		rotL.drawRotRect(simpleTrans.moving[2], this.x-ox, this.y-oy, 16, 10/4, 10/4, fixedrot, torquegraph.getRotationOf(1)+0, torquegraph.getRotationOf(1)+180);
		rotL.drawRotRect(simpleTrans.moving[2], this.x-ox, this.y-oy, 16, 10/4, 10/4, fixedrot, torquegraph.getRotationOf(1)+180, torquegraph.getRotationOf(1)+360);
		
		//
		Draw.rect(simpleTrans.overlaysprite[variant], this.x, this.y, this.rotdeg());
		
		
		//rotL.drawSlideRect(simpleTrans.movingtest, this.x-ox, this.y-oy,26*0.25,9,54*0.25,9,  this.rotdeg(), 18*0.25, this.getNetworkRotation(0)/360.0)
		
		Draw.rect(simpleTrans.topsprite[this.rotation%2], this.x, this.y, 0);
        this.drawTeamTop();

	}
	
	
});

simpleTrans.rotate = true;
simpleTrans.update = true;
simpleTrans.solid = true;
simpleTrans.getGraphConnectorBlock("torque graph").setAccept([2,1,0,0,1,2,0,0]);
simpleTrans.getGraphConnectorBlock("torque graph").setRatio([1,2.5]);
simpleTrans.getGraphConnectorBlock("torque graph").setBaseFriction(0.05);
simpleTrans.getGraphConnectorBlock("torque graph").setBaseInertia(25);

File: main/src/unity/world/blocks/distribution/SimpleTransmission.java
package unity.world.blocks.distribution;

import arc.graphics.g2d.*;
import arc.math.geom.*;
import unity.graphics.*;
import unity.world.blocks.*;

import static arc.Core.atlas;

public class SimpleTransmission extends GraphBlock{
    final TextureRegion[] topRegions = new TextureRegion[2], overlayRegions = new TextureRegion[2], movingRegions = new TextureRegion[3];//topsprite,overlaysprite,moving
    TextureRegion bottomRegion, mbaseRegion;//base,mbase

    public SimpleTransmission(String name){
        super(name);
        rotate = solid = true;
    }

    @Override
    public void load(){
        super.load();
        for(int i = 0; i < 2; i++){
            topRegions[i] = atlas.find(name + "-top" + (i + 1));
            overlayRegions[i] = atlas.find(name + "-overlay" + (i + 1));
        }
        for(int i = 0; i < 3; i++) movingRegions[i] = atlas.find(name + "-moving" + (i + 1));
        bottomRegion = atlas.find(name + "-bottom");
        mbaseRegion = atlas.find(name + "-mbase");
    }

    public class SimpleTransmissionBuild extends GraphBuild{
        @Override
        public void draw(){
            var torqueGraph = torque();
            float graphRot0 = torqueGraph.getRotationOf(0);
            float graphRot1 = torqueGraph.getRotationOf(1);
            float fixedRot = (rotdeg() + 90f) % 180f - 90f;
            int variant = (rotation + 1) % 4 >= 2 ? 1 : 0;
            Draw.rect(bottomRegion, x, y);
            Draw.rect(mbaseRegion, x, y, rotdeg());

            var offset = Geometry.d4(rotation + 1);
            float ox = offset.x * 4f;
            float oy = offset.y * 4f;
            //xelo..
            UnityDrawf.drawRotRect(movingRegions[0], x + ox, y + oy, 16f, 4.5f, 4.5f, fixedRot, graphRot0, graphRot0 + 180f);
            UnityDrawf.drawRotRect(movingRegions[0], x + ox, y + oy, 16f, 4.5f, 4.5f, fixedRot, graphRot0 + 180f, graphRot0 + 360f);

            UnityDrawf.drawRotRect(movingRegions[1], x + ox * -0.125f, y + oy * -0.125f, 16f, 4.5f, 4.5f, fixedRot, 360f - graphRot0, 180f - graphRot0);//360-(a+180)
            UnityDrawf.drawRotRect(movingRegions[1], x + ox * -0.125f, y + oy * -0.125f, 16f, 4.5f, 4.5f, fixedRot, 540f - graphRot0, 360f - graphRot0);//720-(a+180),720-(a+360)

            UnityDrawf.drawRotRect(movingRegions[2], x - ox, y - oy, 16f, 2.5f, 2.5f, fixedRot, graphRot1, graphRot1 + 180f);
            UnityDrawf.drawRotRect(movingRegions[2], x - ox, y - oy, 16f, 2.5f, 2.5f, fixedRot, graphRot1 + 180f, graphRot1 + 360f);

            Draw.rect(overlayRegions[variant], x, y, rotdeg());

            Draw.rect(topRegions[rotation % 2], x, y);
            drawTeamTop();
        }
    }
}
